src/core/ddsi/src/ddsi_typelookup.c

```diff
@@ -138,124 +198,140 @@ void ddsi_tl_meta_ref (struct ddsi_domaingv *gv, const type_identifier_t *type_i
   {
     tlm = ddsrt_malloc (sizeof (*tlm));
     memset (tlm, 0, sizeof (*tlm));
-    tlm->gv = gv;
     tlm->state = TL_META_NEW;
-    tlm->type_id = ddsi_typeid_dup (tid);
+    tlm->type_id = *tid;
     ddsrt_hh_add (gv->tl_admin, tlm);
     GVTRACE (" new %p", tlm);
   }
-  if (src_ep_guid != NULL)
-  {
-    bool found = false;
-    for (uint32_t n = 0; !found && n < tlm->n_endpoints; n++)
-    {
-      if (!memcmp (&tlm->endpoints[n], src_ep_guid, sizeof (*src_ep_guid)))
-        found = true;
-    }
-    if (!found)
-    {
-      tlm->n_endpoints++;
-      tlm->endpoints = ddsrt_realloc (tlm->endpoints, tlm->n_endpoints * sizeof (*tlm->endpoints));
-      memcpy (&tlm->endpoints[tlm->n_endpoints - 1], src_ep_guid, sizeof (*src_ep_guid));
-      GVTRACE (" add ep "PGUIDFMT, PGUID (*src_ep_guid));
-    }
-  }
+  tlm_endpoint_add (gv, tlm, proxy_ep_guid);
```

Comment:
So this call should be conditional on `proxy_ep_guid` being non-null. A few doc comments for the interface exported by this module would be a good idea, too, by the way.